### DISCUSSION TITLE
Kitty create tab with activate a conda env

### DISCUSSION THREAD

**Discussion by zuozuo** [kovidgoyal/kitty#8171] [Q&A] [answered]
When using kitty launch to create a new tab, i want to activate a specific conda env, but i can not make work, can anyone help?

I have tried
```shell
kitty @ launch --type=tab zsh -c 'conda activate aider && exec zsh'
```

this just returns a  number 

![image](https://github.com/user-attachments/assets/9e89522b-c545-41af-8f6d-908af1cd1058)

```shell
kitty @ launch --type=tab zsh -c 'exec zsh && conda activate aider'
```
this successfully creates a new tab, but the conda env does not change.

**Comment by kovidgoyal** [comment, marked as answer]
I dont use conda, but assuming conda activate is meant to run in the
current shell and not create a separate shell, then what you need ot do
is add a section like

```sh
if [[ -n "$CONDA_ACTIVATE" ]]; then conda activate "$CONDA_ACTIVATE"; fi
```

to your zshrc and then in kitty do

kitten @ launch --env CONDA_ACTIVATE=aider

**Reply by zuozuo** [reply]
it works, thanks very much ! @kovidgoyal
